### ISSUE TITLE
I train the model based on vgg-16,but the MAP is not good ,only 0.49.WHY

### ISSUE THREAD

**Issue by guyubit91** [balancap/SSD-Tensorflow#194]
I trained  the model as the instructions the balancap wrote in his page,but the MAP is not good ,only 0.49.WHY

**Comment by HiKapok** [comment]
I wrote a novel detector based on this code, the performance is not very good(<0.5mAP). I guess there may be some same problems with SSD. did you find the key of training using this code successfully?

**Comment by XuanheLiu** [comment]
Is your final map only 0.49? Or before your fine-tuning model? @guyubit91

**Comment by wuguangbin1230** [comment]
@HiKapok Ｈｉ, Do you solve the problem (I wrote a novel detector based on this code, the performance is not very good(<0.5mAP)?

**Comment by HiKapok** [comment]
@wuguangbin1230 I made some improvement but that's another problem, besides I have also found that the initializer matters because the new added layers may be trained longer. Maybe you can try use a truncated normal initializer with larger variance, eg:0.005. This can reduce my training loss a lot, but I don't know if that works for you. I think you can check the original SSD code in caffe to get better performance. Good luck..

**Comment by wuguangbin1230** [comment]
@HiKapok 　Ｔｈank you for  your reply! What is your performance now(mAP)?
Do you know whether the function (mask = tf.logical_and(mask, feat_scores > -0.5)) ｉｓ right in ssd_common.py? Or mask = tf.logical_and(mask, feat_scores > 0.5)

**Comment by HiKapok** [comment]
@wuguangbin1230 I also have a few puzzles about the code. Most of them had been fixed by myself. For example: the preprocess method didn't match the SSD paper, and so did the anchors matching strategy, so sometimes result in NAN in loss. You can refer to one of my repositories called [RON_Tensorflow](https://github.com/HiKapok/RON_Tensorflow) for those fixes. The performance for my novel detector(not this SSD) is 0.56mAP before I fixed one silly bugs.

**Comment by wuguangbin1230** [comment]
@HiKapok Ok！ Thank ｙｏｕ！

**Comment by HiKapok** [comment]
@wuguangbin1230 Hi, I fixed some other bugs just now, I think these fixes matter a lot, you can refer to my code if you don't have ideas about how to improv this code.

**Comment by HiKapok** [comment]
If anyone couldn't train this codes successfully, you can try [this one](https://github.com/HiKapok/SSD.TensorFlow)

**Comment by alexekkis** [comment]
This TensorFlow-based Keras port achieves the same mAP as the original Caffe implementation when trained from scratch: https://github.com/pierluigiferrari/ssd_keras

**Comment by luojiapeng14** [comment]
@HiKapok Thanks a lot. I have stared you SSD repo, It works well.  I'm trying to train SSD on widerface dataset. Before that, I waste a lot of time on this implement. I'll very appreciate if you can inform me where the mistake is in this implementation because it really take me a lot of time and I didn't work it out.
